### PR TITLE
hls4ml profiling

### PR DESCRIPTION
**Pull request by Chriisbrown**
This PR adds hls4ml profiling functionality where each hls4ml layer is individually checked for agreement with the keras model, this has been added with a new script and new CI job

It also switches the softmax implementation to "latency" due to the increased performance of using this implementation

Minor changes include updating the CI to store physics and profiling plots correctly

### CODE REVIEW

File: tagger/firmware/hls4ml_profile.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+matplotlib.rcParams['xtick.major.size'] = 20
+matplotlib.rcParams['xtick.major.width'] = 5
+matplotlib.rcParams['xtick.minor.size'] = 10
+matplotlib.rcParams['xtick.minor.width'] = 4
+
+matplotlib.rcParams['ytick.major.size'] = 20
+matplotlib.rcParams['ytick.major.width'] = 5
+matplotlib.rcParams['ytick.minor.size'] = 10
+matplotlib.rcParams['ytick.minor.width'] = 4
```
Comment (Duchstf) [inline tagger/firmware/hls4ml_profile.py:57]: We should refactor this in the future, but probably not needed for the current PR

File: CI/ml_pipeline.yml [hunk 2]
```diff
@@ -147,24 +147,54 @@ emulation-evaluate:
...
     - export PYTHONPATH=/builds/ml_l1/TrainTagger:$PYTHONPATH
   script:
     - mkdir $Name
     - mv tagger/firmware $Name
     - mv output/baseline/model $Name/model
     - mv output/baseline/plots/emulation $Name/plots
+    - mv output/baseline/plots/profile $Name/plots
     - mv output/baseline/plots/training/ $Name/plots
-    #- mv output/baseline/plots/testing/ $Name/plots
+    - mv output/baseline/plots/physics/ $Name/plots
```
Comment (Duchstf) [inline CI/ml_pipeline.yml:197]: Feel like we need more long term solution for this? maybe it's better to just move everything in `output/baseline/plots/`????

File: tagger/firmware/hls4ml_convert.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def convert(model, outpath,build=True):
 
     
     config["LayerName"]["jet_id_output"]["Precision"]["result"] = class_precision
-    config["LayerName"]["jet_id_output"]["Implementation"] = "legacy"
+    config["LayerName"]["jet_id_output"]["Implementation"] = "latency"
```
Comment (Duchstf) [inline tagger/firmware/hls4ml_convert.py:48]: Are the synthesized results saved somewhere? I guess it can be downloaded as a CERN Gitlab artifact for now?

**Review by Duchstf — APPROVED** [review]

File: CI/ml_pipeline.yml (hunk 2 above)
Comment (Chriisbrown) [reply CI/ml_pipeline.yml:197]: If we do that the website will just be a big page of plots making it quite hard to go through them, but there probably is a better way than this where we can automate the creation of folders before being uploaded
